### DISCUSSION TITLE
Setting axis range?

### DISCUSSION THREAD

**Discussion by Hvass-Labs** [observablehq/plot#1938] [Q&A] [answered]
Two related questions:

1) I am making a simple line-plot with dates on the x-axis and values on the y-axis. The values range between e.g. 20 and 40, so Plot adapts the y-axis to that range, but I want the y-axis to begin already at zero. How do I manually set the y-axis range e.g. to start at zero?

2) Sometimes the data contains negative values as well e.g. down to -30. In this case I want to keep the plot's default range for the y-axis and not set it to start at zero. But I want to use the exact same code to generate the plot, because my data comes from a database. How do I change the y-axis range to e.g. the minimum of its default range and some other value such as zero?

Thanks!

**Comment by Fil** [comment, marked as answer]
1. use the `y: {zero: true}` option, or add a mark that includes 0 as a value for y (typically, `Plot.ruleY([0])`).
2. same answer :)

**Reply by Hvass-Labs** [reply]
`y: {zero: true}` is an elegant solution that works in my case where I want to ensure zero is on the y-axis. But for an arbitrary value `a` you suggest using `Plot.ruleY([a])`, which adds a horizontal line that I don't want. How can I avoid that? Thanks.

**Reply by Fil** [reply]
`Plot.ruleY([a], {filter: false})`? It will add *a* to the domain, but won't display anything.
See also https://github.com/observablehq/plot/issues/1577

**Reply by Hvass-Labs** [reply]
OK, thanks. It's hacky but it works :-)
